### DISCUSSION TITLE
Ignore Gradle/Maven test dependencies

### DISCUSSION THREAD

**Discussion by Mcluky** [renovatebot/renovate#8197] [Request Help] [answered]
Hi

I'm trying to figure out how I can configure Renovate so that it does not create pull requests to update Maven or Gradle test-scoped dependencies. I've seen that there is an option to do something similar for npm dependencies using the `devDependencies` type like the following;
```json
{
  "packageRules": [
    {
      "depTypeList": [
        "devDependencies"
      ],
      "enabled": false
    }
  ]
}
```
Unfortunately, this does not work for Maven and Gradle. Can I configure this somehow differently? I wasn't able to find a setting in the documentation that would help me achieve this.

**Comment by rarkins** [comment]
It looks like Renovate doesn't classify dependencies by type in maven. Can you provide an example `pom.xml` of how/where they're defined?

**Reply by Mcluky** [reply]
Hi, thanks for the quick response. The scope of a dependency is defined in the `<scope>` attribute in maven. A basic `pom.xml` could look like the following:

```xml
<project xmlns="http://maven.apache.org/POM/4.0.0"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
  <modelVersion>4.0.0</modelVersion>

  <groupId>com.some.package</groupId>
  <artifactId>applicationname</artifactId>
  <version>0.4.2</version>
  <dependencies>
    <!-- Normal dependency -->
    <dependency>
        <groupId>com.google.guava</groupId>
        <artifactId>guava</artifactId>
        <version>30.1-jre</version>
    </dependency>
    <!-- Test dependency only available during tests, not in productive code -->
    <dependency>
        <groupId>org.mockito</groupId>
        <artifactId>mockito-all</artifactId>
        <version>1.10.19</version>
        <scope>test</scope>
    </dependency>
  </dependencies>
</project>
```

**Comment by rarkins** [comment, marked as answer]
Resolution: this will require implementation of this feature request: https://github.com/renovatebot/renovate/issues/8198
